lib/nanoc/base/compilation/compiler.rb

```diff
@@ -72,14 +72,14 @@ class Compiler
 
     # @group Public instance methods
 
-    def initialize(site, rules_collection, params = {})
+    def initialize(site, rules_collection, compiled_content_cache:, checksum_store:, rule_memory_store:, rule_memory_calculator:)
```

Comment:
Oh, I didn't realize keyword args could be empty. I've always defaulted them to something.